### ISSUE TITLE
Feature: FPDI PDF 1.5+ Support

### ISSUE THREAD

**Issue by jakejackson1** [mpdf/mpdf#28]
Gents, since it is clear the [open source version of FDPI](https://github.com/mpdf/mpdf/pull/16) won't support the ability to read PDF 1.5+ in the near future I've been looking into the "hows" of developing an open source version the whole community can use (it involved a lot of laborious reading of the PDF spec).

I believe the current blockers are two-fold:
1. The support of new filters introduced in PDF 1.5. 
   
   ASCIIHexDecode - Already Support in FPDI
   ASCII85Decode - Already Support in FPDI
   LZWDecode - Already Support in FPDI
   **FlateDecode - Need to Implement**
   **RunLengthDecode - Need to Implement**
   CCITTFaxDecode - Not sure about this yet
   JBIG2Decode - Not sure about this yet
   DCTDecode - Not sure about this yet
   JPXDecode - Not sure about this yet
   Crypt - Not sure about this yet
   
   Assuming the [licensing issue is resolved](https://github.com/mpdf/mpdf/issues/15), we'll be able to borrow heavily from the TCPDF project which actually [implements the _FlateDecode_ and _RunLengthDecode_ filters already](https://github.com/tecnickcom/TCPDF/blob/develop/include/tcpdf_filters.php). That makes that job a bit easier. 
2. FPDI currently only supports standard xref tables. PDF 1.6 introduced xref cross reference streams which will need to be handled correctly. Again, the good news is the TCPDF project also implements this correctly in the `getXrefData` / `decodeXrefStream` / `decodeXref` methods in their [tcpdf_parser.php file](https://github.com/tecnickcom/TCPDF/blob/develop/tcpdf_parser.php).

There could be additional blockers preventing this feature being implemented but I won't know without some real-world testing. 

This isn't currently a high priority, but I feel it's one of those must-have features the PHP PDF open-source community will greatly benefit from.

I'll be playing around with this possibility in my free time. If anyone would like to contribute it would be greatly appreciated.

**Comment by sztyup** [comment]
Is there any chance this is going to be implemented in the foreseeable future?

**Comment by jakejackson1** [comment]
@sztyup no. This functionality isn't high on my priority list, but others are more than welcome to contribute.

**Comment by sonicpunk** [comment]
@jakejackson1 We are also needing to support PDFs 1.5+ and are willing to purchase a license for this. Do you know the steps involved in how to install the license and would mpdf still be compatible?

**Comment by dbfischer** [comment]
> [@jakejackson1](https://github.com/jakejackson1) We are also needing to support PDFs 1.5+ and are willing to purchase a license for this. Do you know the steps involved in how to install the license and would mpdf still be compatible?

This question is nearly 3 years old and we also like to have support for PDFs 1.5+. Is there an answer for it?

**Comment by dbfischer** [comment]
How do I use the license with mPDF? Do I have to copy files from the licenced version of  FPDI to mPDF? Will it work that easy?
